src/docfx/build/DocsetBuilder.cs

```diff
@@ -0,0 +1,186 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Concurrent;
+using System.Diagnostics.CodeAnalysis;
+using System.Threading.Tasks;
+
+namespace Microsoft.Docs.Build
+{
+    internal class DocsetBuilder
+    {
+        private readonly string _workingDirectory;
+        private readonly string _docsetPath;
+        private readonly string? _outputPath;
+        private readonly CommandLineOptions _options;
+
+        public DocsetBuilder(string workingDirectory, string docsetPath, string? outputPath, CommandLineOptions options)
+        {
+            _workingDirectory = workingDirectory;
```

Comment:
As from the work item description
- the member fields can be shared for different builds, which are readonly, **but these are just docset input?**
- dependencies created on-the-fly are not readonly, **but are they still be shared between file build?**